pkg/nats/subscriber.go

```diff
@@ -62,74 +42,95 @@ type StreamingSubscriberConfig struct {
 	// It is mapped to stan.AckWait option.
 	AckWaitTimeout time.Duration
 
-	// StanOptions are custom []stan.Option passed to the connection.
-	// It is also used to provide connection parameters, for example:
-	// 		stan.NatsURL("nats://localhost:4222")
-	StanOptions []stan.Option
+	// SubscribeTimeout determines how long subscriber will wait for a successful subscription
+	SubscribeTimeout time.Duration
 
-	// StanSubscriptionOptions are custom []stan.SubscriptionOption passed to subscription.
-	StanSubscriptionOptions []stan.SubscriptionOption
+	// NatsOptions are custom []nats.Option passed to the connection.
+	// It is also used to provide connection parameters, for example:
+	// 		nats.URL("nats://localhost:4222")
+	NatsOptions []nats.Option
 
 	// Unmarshaler is an unmarshaler used to unmarshaling messages from NATS format to Watermill format.
 	Unmarshaler Unmarshaler
-}
 
-type StreamingSubscriberSubscriptionConfig struct {
-	// StanSubscriptionOptions are custom []stan.SubscriptionOption passed to subscription.
-	StanSubscriptionOptions []stan.SubscriptionOption
+	// SubjectCalculator is a function used to transform a topic to an array of subjects on creation (defaults to "{topic}.*")
+	SubjectCalculator SubjectCalculator
+
+	// AckSync enables synchronous acknowledgement (needed for exactly once processing)
+	AckSync bool
```

Comment:
By principle, all default configurations of Watermill should guarantee the biggest consistency (with the tradeoff of performance). 

Do you think that in that case, it may make sense to rename this configuration option to AckAsync, so by default it would guarantee higher consistency?

Maybe it would also make sense to name it ExactlyOnceDeliveryAckAsync, to surface that it works just in this context.

Reply:
Probably should default to using jetstream too

Reply:
done see https://github.com/ThreeDotsLabs/watermill-nats/pull/8/commits/9525332962990be4978539f761be2796c73688c2

Reply:
👍